src/Manifolds.jl

```diff
@@ -102,6 +106,94 @@ function hat(M::Manifold, x, vⁱ)
     return v
 end
 
+@doc doc"""
+    mean(M,x; weights=1/n*ones(n), stop_tol=10^-7, stop_iter=100 )
+
+computes the Riemannian center of mass also known as Karcher mean of the vector
+`x` of `n` points on the [`Manifold`](@ref) `M`. This function
+uses the gradient descent scheme. Optionally one can provide
+weights $w_i$ for the weighted Riemannian center of mass.
+The general formula to compute the minimizer reads
+````math
+\argmin_{y\in\mathcal M} \frac{1}{2}\sum_{i=1}^n w_i\mathrm{d}_{\mathcal M}^2(y,x_i),
+````
+where $\mathrm{d}_{\mathcal M}$ denotes the Riemian [`distance`](@ref).
+
+Finally `stop_tol` denotes the minimal change between to iterates then to stop
+and `stop_iter` the maximal number of iterations. For more stopping criteria
+check the [`Manopt.jl`](https://manoptjl.org) package and use a solver therefrom.
+
+The algorithm is further described in 
+> Afsari, B; Tron, R.; Vidal, R.: On the Convergence of Gradient
+> Descent for Finding the Riemannian Center of Mass,
+> SIAM Journal on Control and Optimization (2013), 51(3), pp. 2230–2260, 
+> doi: [10.1137/12086282X](https://doi.org/10.1137/12086282X),
+> arxiv: [1201.0925](https://arxiv.org/abs/1201.0925">1201.0925)
+"""
+function mean(M::Manifold, x::Vector{T};
```

Comment:
I'd use abstract type here `x::AbstractVector`.

Reply:
And we don't need `T`